src/g_game.c

```diff
@@ -1171,6 +1171,58 @@ int G_GotoNextLevel(int *pEpi, int *pMap)
   return false;
 }
 
+int G_GotoPrevLevel(void)
+{
+    if (gamestate != GS_LEVEL || deathmatch || netgame || demorecording
+        || demoplayback || menuactive)
+    {
+        return false;
+    }
+
+    const int cur_epsd = gameepisode;
+    const int cur_map = gamemap;
+    int ret = false;
+
+    do
```

Comment:
It looks like there are issues with Legacy of Rust (both next and prev directions).

Reply:
The latest commit should fix this.

I couldn't spot a problem with the next key, though. If you mean that each map that ends the game continues at E1M1/MAP01, that's intended.

Reply:
Here's the current behavior:

Start first episode:
- Press "next map": repeats e1m1-e1m7 forever
- Press "prev map": initially e1m1, then e2m7-e2m1, then "previous level not found for map08" (stays on e2m1)

Start second episode:
- Press "next map": e2m1-e2m7, then repeats e1m1-e1m7 forever
- Press "prev map": initially e2m1, then "previous level not found for map08" (stays on e2m1)

Is that intended? It seems kind of odd.

Reply:
I confirm that Legacy of Rust works strangely. I suggest to remove "cycle" - when user presses "prev" on first map, last map is not searched. Episode concept in UMAPINFO makes it overcomplicated and confusing for user. I think we should remove "cycle" from "next map" as well, for consistency.

Reply:
Alright, so should I modify the code to remove cycling from prev/next key and let the sequence end at maps that end the game?

Reply:
> Alright, so should I modify the code to remove cycling from prev/next key and let the sequence end at maps that end the game?

Yes, I think it will be right.

Reply:
> Yes, I think it will be right.

done

Reply:
Everything works nicely now.